modyn/tests/evaluator/internal/grpc/test_evaluator_grpc_servicer.py

```diff
@@ -339,6 +340,35 @@ def test_setup_metrics(test_connect_to_model_storage, test_connect_to_storage):
         assert isinstance(metrics[0], Accuracy)
 
 
+@patch.object(EvaluatorGRPCServicer, "connect_to_storage", return_value=DummyStorageStub())
+@patch.object(EvaluatorGRPCServicer, "connect_to_model_storage", return_value=DummyModelStorageStub())
+def test_setup_metrics_multiple_f1(test_connect_to_model_storage, test_connect_to_storage):
+    with tempfile.TemporaryDirectory() as modyn_temp:
+        evaluator = EvaluatorGRPCServicer(get_modyn_config(), pathlib.Path(modyn_temp))
+
+        macro_f1_config = MetricConfiguration(
+            name="F1Score",
+            config=JsonString(value=json.dumps({"num_classes": 2, "average": "macro"})),
+            evaluation_transformer=PythonString(value=""),
+        )
+
+        micro_f1_config = MetricConfiguration(
+            name="F1Score",
+            config=JsonString(value=json.dumps({"num_classes": 2, "average": "micro"})),
+            evaluation_transformer=PythonString(value=""),
+        )
+        # not double macro, but macro and micro work
+        metrics = evaluator._setup_metrics([macro_f1_config, micro_f1_config, macro_f1_config])
+
+        assert len(metrics) == 2
+        assert isinstance(metrics[0], F1Score)
+        assert isinstance(metrics[1], F1Score)
+        assert metrics[0].average == F1ScoreTypes.MACRO
```

Comment:
Mostly personal preference, but I think literal types are better suited here as they are more lightweight and support the same IDE convenience. I only use enum types if I need some way to define extra functionality through member functions.

Reply:
i guess i did not really think too much here. when i refactor this to using the pydantic stuff, I'll probably choose literals :D